core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/new_impl/RawInfraFromRjsBuilder.kt

```diff
@@ -0,0 +1,822 @@
+            trackSectionOffsetChunks[chunk.offset.distance] = chunkIdx
+        }
+        trackSectionDistanceSortedChunkMap[trackSectionIdx] = trackSectionOffsetChunks
+        return trackSectionIdx
+    }
+
+    fun trackChunk(
+        geo: LineString,
+        slopes: DirectionalMap<DistanceRangeMap<Double>>,
+        curves: DirectionalMap<DistanceRangeMap<Double>>,
+        gradients: DirectionalMap<DistanceRangeMap<Double>>,
+        length: Length<TrackChunk>,
+        offset: Offset<TrackSection>,
+        loadingGaugeConstraints: DistanceRangeMap<LoadingGaugeConstraint>
+    ): TrackChunkId {
+        return trackChunkPool.add(
+            TrackChunkDescriptor(
+                // TODO: The track ID will be filled later, track is not initialized yet
+                StaticIdx(0u),
+                offset,
+                length,
+                geo,
+                slopes,
+                curves,
+                gradients,
+                // Route IDs will be filled later on, routes are not initialized yet
+                DirectionalMap(MutableStaticIdxArrayList(), MutableStaticIdxArrayList()),
+                // OperationalPointPart IDs will be filled later on, operational point parts
+                // are not initialized yet
+                MutableStaticIdxArrayList(),
+                loadingGaugeConstraints,
+                // Electrifications will be filled later on
+                distanceRangeMapOf(
+                    listOf(DistanceRangeMap.RangeMapEntry(0.meters, length.distance, ""))
+                ),
+                // NeutralSections will be filled later on
+                DirectionalMap(distanceRangeMapOf(), distanceRangeMapOf()),
+                // SpeedSections will be filled later on
+                DirectionalMap(
+                    distanceRangeMapOf(
+                        listOf(
+                            DistanceRangeMap.RangeMapEntry(
+                                0.meters,
+                                length.distance,
+                                SpeedSection(
+                                    Double.POSITIVE_INFINITY.metersPerSecond,
+                                    mapOf(),
+                                    mapOf()
+                                )
+                            )
+                        )
+                    ),
+                    distanceRangeMapOf(
+                        listOf(
+                            DistanceRangeMap.RangeMapEntry(
+                                0.meters,
+                                length.distance,
+                                SpeedSection(
+                                    Double.POSITIVE_INFINITY.metersPerSecond,
+                                    mapOf(),
+                                    mapOf()
+                                )
+                            )
+                        )
+                    )
+                )
+            )
+        )
+    }
```

Comment:
Huge nit, but this is a 50 lines long statement going ~8 levels of indentation deep